contracts/src/emergency/sequencer_uptime_feed.cairo

```diff
@@ -0,0 +1,358 @@
+
+    #[event]
+    fn L1SenderTransferred(from_address: u256, to_address: u256) {}
+
+    impl Aggregator of IAggregator {
+        fn latest_round_data() -> Round {
+            require_access();
+            let latest_round_id = _latest_round_id::read();
+            let round_transmission = _round_transmissions::read(latest_round_id);
+            // bool -> felt252 -> u128
+            let answer: u128 = round_transmission.answer.into().try_into().unwrap();
+            Round {
+                round_id: latest_round_id.into(),
+                answer: answer,
+                block_num: round_transmission.block_num,
+                started_at: round_transmission.started_at,
+                updated_at: round_transmission.updated_at,   
+            }
+        }
+
+        fn round_data(round_id: u128) -> Round {
+            require_access();
+            assert(round_id < _latest_round_id::read(), 'invalid round id');
+            let round_transmission = _round_transmissions::read(round_id);
+            // bool -> felt252 -> u128
+            let answer: u128 = round_transmission.answer.into().try_into().unwrap();
+            Round {
+                round_id: round_id.into(),
+                answer: answer,
+                block_num: round_transmission.block_num,
+                started_at: round_transmission.started_at,
+                updated_at: round_transmission.updated_at,   
+            }
+        }
+
+        fn description() -> felt252 {
+            'L2 Sequencer Uptime Status Feed'
+        }
+
+        fn decimals() -> u8 {
+            0_u8
+        }
+
+        fn type_and_version() -> felt252 {
+            'SequencerUptimeFeed 1.0.0'
+        }
+    }
+
+    #[constructor]
+    fn constructor(initial_status: bool, owner_address: ContractAddress) {
+        initializer(initial_status, owner_address);
+    }
+
+    #[l1_handler]
+    fn update_status(from_address: felt252, status: bool, timestamp: u64) {
+        assert(_l1_sender::read() == from_address.into(), 'EXPECTED_FROM_BRIDGE_ONLY');
+
+        let latest_round_id = _latest_round_id::read();
+        let latest_round = _round_transmissions::read(latest_round_id);
+
+        if (timestamp <= latest_round.started_at) {
+            UpdateIgnored(latest_round.answer, latest_round.started_at, status, timestamp);
+            return ();
+        }
+
+        if (latest_round.answer == status) {
+            _update_round(latest_round_id, latest_round);
+        } else {
+            // only increment round when status flips
+            let round_id = latest_round_id + 1_u128;
+            _record_round(round_id, status, timestamp);
+        }
+    }
+
+    #[external]
+    fn set_l1_sender(address: u256) {
+        Ownable::assert_only_owner();
+        // because u256 literals are not supported
+        let ETH_ADDRESS_BOUND = u256 { high: 0x100000000_u128, low: 0_u128 }; // 2 ** 160
```

Comment:
160 - 128 = 32. So there should be 32 zero bits on the high side. Each hex digit is 4 bits. So 32/4 = 8 hex digits should be 0